keras/src/distribution/tensor_parallel/tensor_layout.py

```diff
@@ -0,0 +1,67 @@
+import collections
+
+from keras.src import ops
+
+
+class Split:
+    """Splits a tensor into shards along a specified dimension.
+
+    This is an internal utility used by a higher-level distribution API.
+    It implements sharding by slicing a tensor along one of its axes.
+    It handles cases where the dimension size is not perfectly divisible by the
+    number of workers by distributing the remainder elements one by one to the
+    first few workers.
+    """
+
+    def __init__(self, device_count, dim, sharding_type="auto"):
+        """Initializes the Split action.
+
+        Args:
+            device_count: The total number of workers/shards.
+            dim: The dimension along which to split the tensor. If -1, the
+                last dimension is used.
+            sharding_type: If `dim` is -1, this can be 'row' (dim=0) or
+                'column' (dim=1) to infer the split axis for 2D tensors.
+                Defaults to "auto".
+        """
+        self.device_count = device_count
+        self.dim = dim
+        self.sharding_type = sharding_type
+
+        if dim == -1 and sharding_type != "auto":
+            if sharding_type == "row":
+                self.dim = 0
+            elif sharding_type == "column":
+                self.dim = 1
+
+    def __call__(self, tensor, rank):
+        """Splits the tensor and returns the shard corresponding to the rank.
+
+        This method calculates the correct slice of the tensor for a given
+        worker rank, handling uneven distributions gracefully.
+
+        Args:
+            tensor: The full tensor to be sharded.
+            rank: The rank of the worker for which to get the shard.
+
```

Comment:
`rank` is a bit confusing here, normally we only use the term for the rank of a tensor. How about `index`?

Same comment about the term worker. Just say `index: The index of the slice to return`.